autoload/lsp/util.vim

```diff
@@ -85,7 +85,12 @@ export def LspUriToFile(uri: string): string
   if uri_decoded =~? '^file:///\a:'
     # MS-Windows URI
     uri_decoded = uri_decoded[8 : ]
-    uri_decoded = uri_decoded->substitute('/', '\\', 'g')
+    if has("win32unix")  # we're in Cygwin
+      # The substitution is to remove the '^@' escape character from the end of line.
+      uri_decoded = system($'cygpath --unix {uri_decoded}')->substitute('^\(\p*\).*$', '\=submatch(1)', "")
```

Comment:
Invoking an external command for decoding every URI may incur a lot of overhead.  Is cygpath used to only the '^@' character from the end of the line?  Is it possible to remove it using trim()?